geode-redis/src/distributedTest/java/org/apache/geode/redis/RedisSessionDistDUnitTest.java

```diff
@@ -0,0 +1,395 @@
+    redisPropsForServer1.setProperty("redis-port", Integer.toString(server1Port));
+    redisPropsForServer1.setProperty("log-level", "warn");
+
+    redisPropsForServer2 = new Properties();
+    redisPropsForServer2.setProperty("redis-bind-address", "localHost");
+    redisPropsForServer2.setProperty("redis-port", Integer.toString(server2Port));
+    redisPropsForServer2.setProperty("log-level", "warn");
+
+    cluster.startServerVM(indexOfServer1, redisPropsForServer1, locator.getPort());
+    cluster.startServerVM(indexOfServer2, redisPropsForServer2, locator.getPort());
+
+    client1 = cluster.getVM(indexOfClient1);
+    client2 = cluster.getVM(indexOfClient2);
+
+    client1.invoke("start a spring app", () -> {
+      System.setProperty("server.port", Integer.toString(localClient1Port));
+      System.setProperty("spring.redis.port", Integer.toString(localServer1Port));
+      SpringApplication.run(
+          RedisSpringTestApplication.class,
+          "" + localServer1Port, "" + localServer2Port);
+    });
+
+    client2.invoke("start a spring app", () -> {
+      System.setProperty("server.port", Integer.toString(localClient2Port));
+      System.setProperty("spring.redis.port", Integer.toString(localServer2Port));
+      springApplicationContext = SpringApplication.run(
+          RedisSpringTestApplication.class,
+          "" + localServer1Port, "" + localServer2Port);
+    });
+
+    restTemplate = new RestTemplate();
+    jedis = new Jedis("localHost", server1Port, JEDIS_TIMEOUT);
+  }
+
+  @After
+  public void cleanupAfterTest() {
+    jedis.flushAll();
+  }
+
+  @AfterClass
+  public static void cleanupAfterClass() {
+    jedis.disconnect();
+  }
+
+  @Test
+  public void should_beAbleToCreateASession() {
+    HttpEntity<String> request = new HttpEntity<>("note1");
+    cluster.getMember(0);
+    HttpHeaders resultHeaders = restTemplate
+        .postForEntity(
+            LOCALHOST + ":" + client1Port + "/addSessionNote",
+            request,
+            String.class)
+        .getHeaders();
+
+    assertThat(resultHeaders).isNotNull();
+
+    String cookieString = resultHeaders.get("Set-Cookie").get(0);
+    HttpHeaders requestHeaders = new HttpHeaders();
+    requestHeaders.add("Cookie", cookieString);
+
+    List<HttpCookie> cookies = HttpCookie.parse(cookieString);
+    byte[] decodedCookie = Base64.getDecoder().decode(cookies.get(0).getValue());
+
+    assertThat(jedis.hgetAll("spring:session:sessions:" + new String(decodedCookie))).isNotEmpty();
+  }
+
+  @Test
+  public void should_storeSessionDataInRedis() {
+    HttpEntity<String> request = new HttpEntity<>("note1");
+    HttpHeaders responseHeaders = restTemplate
+        .postForEntity(
+            LOCALHOST + ":" + client1Port + "/addSessionNote",
+            request,
+            String.class)
+        .getHeaders();
+
+    String sessionAsCookie = responseHeaders.get("Set-Cookie").get(0);
+
+    HttpHeaders requestHeaders = new HttpHeaders();
```

Comment:
These blocks processing HttpHeaders and HttpCookies could be put into methods; they are just tools to access the data, not material to what we're testing themselves.